salt/states/pkg.py

```diff
@@ -1448,6 +1482,14 @@ def installed(
 
     kwargs['saltenv'] = __env__
     refresh = salt.utils.pkg.check_refresh(__opts__, refresh)
+
+    # check if capabilities should be checked and modify the requested packages
+    # accordingly.
+    if pkgs:
+        (pkgs, was_refreshed) = _resolve_capabilities(pkgs, refresh=refresh, **kwargs)
+        if was_refreshed:
+            refresh = False
```

Comment:
```python
refresh = not was_refreshed
```
...perhaps?

Reply:
No, it would not be the same.

In case `refresh` is already false, `was_refreshed` will also be false and `refresh` would stay as false.
You code would turn `refresh` into true in this case.

Reply:
Ok. Simply `return pkg, False`. :wink: